commit 0c8c5e8679ce2f152a5127254d8f8d4b73c0a2bb
Author: Vladimir Kondratiev <wulf@cicgroup.ru>
Date:   Mon Jul 27 20:48:58 2015 +0300

    Set NOTE_EXTEND flag on directory change if that change is renaming
    a file inside this directory
    
    Required by libinotify-kqueue library

diff --git a/sys/kern/vfs_subr.c b/sys/kern/vfs_subr.c
index 4df8de8..a5518b9 100644
--- sys/kern/vfs_subr.c
+++ sys/kern/vfs_subr.c
@@ -4365,8 +4365,8 @@ vop_rename_post(void *ap, int rc)
 	struct vop_rename_args *a = ap;
 
 	if (!rc) {
-		VFS_KNOTE_UNLOCKED(a->a_fdvp, NOTE_WRITE);
-		VFS_KNOTE_UNLOCKED(a->a_tdvp, NOTE_WRITE);
+		VFS_KNOTE_UNLOCKED(a->a_fdvp, NOTE_WRITE | NOTE_EXTEND);
+		VFS_KNOTE_UNLOCKED(a->a_tdvp, NOTE_WRITE | NOTE_EXTEND);
 		VFS_KNOTE_UNLOCKED(a->a_fvp, NOTE_RENAME);
 		if (a->a_tvp)
 			VFS_KNOTE_UNLOCKED(a->a_tvp, NOTE_DELETE);
